runtime/runtime.go

```diff
@@ -35,6 +36,7 @@ type Runtime struct {
 	logger         *zap.Logger
 	activity       activity.Client
 	metastore      drivers.Handle
+	catalog        drivers.Handle
```

Comment:
It feels like this should be at the instance-level instead, since the catalog is instance-specific, i.e. passing a `CatalogConnector` to `CreateInstance`

Reply:
For cloud `metastore` acts as `catalog` as well (when not using `embedCatalog`) which is not instance specific. If I keep this at instance level then we need to do some changes in `connnection.go` to make sure all instances share the same handle or use instance specific `catalog` for cloud as well (which seems unnecessary).

Reply:
We can keep the current behavior where it falls back to the metastore if `catalogConnector == ""`. That should avoid any migrations or significant refactors.

Eventually we should probably just set `catalogConnector = "metastore"` on the instances, but yes that requires a refactor enabling them to access system connectors, we can skip on that now.